src/reflection.cpp

```diff
@@ -481,4 +481,232 @@ Offset<const Table *> CopyTable(FlatBufferBuilder &fbb,
   }
 }
 
+bool VerifyStruct(flatbuffers::Verifier &v,
+                  const flatbuffers::Table &parentTable,
+                  voffset_t fieldOffset,
+                  const reflection::Object &obj,
+                  bool isRequired) {
+  assert(obj.is_struct());
+  auto offset = parentTable.GetOptionalFieldOffset(fieldOffset);
+  if (!isRequired) {
```

Comment:
Because of the nature of the check we would always need an if/else either inside or outside the function. Currently VerifyStruct and VerifyVectorOfStructs can handle both required and optional fields. If I make this change then they would only be able to handle required fields. And the place where I would be iterating the fields would look like
```
auto offset = table->GetOptionalFieldOffset(fieldDef->offset());
if (!offset && fieldDef->required()) {
  return false;
}

if (!VerifyStruct(v, *table, fieldDef->offset(), *childObj)) {
  return false;
}
```

Which is not very pretty either. If you really want it I can make this change but my preference would be to encapsulate this check inside the function like it is right now.